src/EOFeedManager.sol

```diff
@@ -180,7 +182,19 @@ contract EOFeedManager is IEOFeedManager, OwnableUpgradeable {
     function _processVerifiedRate(bytes memory data, uint256 blockNumber) internal {
         (uint16 feedId, uint256 rate, uint256 timestamp) = abi.decode(data, (uint16, uint256, uint256));
         if (!_supportedFeedIds[feedId]) revert FeedNotSupported(feedId);
-        if (_priceFeeds[feedId].timestamp >= timestamp) revert SymbolReplay(feedId);
+        // if timestamp is greater than 2099-01-01, it means the timestamp is in milliseconds
+        if (timestamp > MAX_SECONDS_TIMESTAMP) {
+            timestamp = timestamp / 1000;
+        }
+
+        uint256 lastTimestamp = _priceFeeds[feedId].timestamp;
+        // auto fix milliseconds timestamp in storage
+        if (lastTimestamp > MAX_SECONDS_TIMESTAMP) {
+            lastTimestamp = lastTimestamp / 1000;
+            _priceFeeds[feedId].timestamp = lastTimestamp;
```

Comment:
we set here lastTimestamp for` _priceFeeds[feedId] ` but in line 198 - we set it again to timestamp
`        _priceFeeds[feedId] = PriceFeed(rate, timestamp, blockNumber);`